src/lib/dnssd/Resolver_ImplMinimalMdns.cpp

```diff
@@ -452,18 +452,49 @@ void MinMdnsResolver::AdvancePendingResolverStates()
         else if (resolver->IsActiveOperationalParse())
         {
             MATTER_TRACE_SCOPE("Active operational delegate call", "MinMdnsResolver");
-            ResolvedNodeData nodeData;
+            ResolvedNodeData nodeResolvedData;
+            CHIP_ERROR err = CHIP_ERROR_UNINITIALIZED;
+
+            if (mActiveResolves.HasBrowseFor(chip::Dnssd::DiscoveryType::kCommissionableNode))
+            {
+                DiscoveredNodeData nodeData;
+                err = resolver->Take(nodeData);
+                if (!nodeData.Is<OperationalNodeData>())
+                {
+                    err = CHIP_ERROR_INVALID_ARGUMENT;
+                }
+                if (err != CHIP_NO_ERROR)
+                {
+                    ChipLogError(Discovery, "Failed to take discovery result: %" CHIP_ERROR_FORMAT, err.Format());
+                }
+                if (mDiscoveryContext != nullptr && err == CHIP_NO_ERROR)
+                {
+                    mDiscoveryContext->OnNodeDiscovered(nodeData);
+                    auto opNodeData                  = nodeData.Get<OperationalNodeData>();
+                    nodeResolvedData.resolutionData  = static_cast<CommonResolutionData>(opNodeData);
+                    nodeResolvedData.operationalData = opNodeData;
+                }
+                else
+                {
+#if CHIP_MINMDNS_HIGH_VERBOSITY
+                    ChipLogError(Discovery, "No delegate to report commissioning node discovery");
```

Comment:
This log does not look correct; we might have a delegate, but `err` could be a failure.

Reply:
fixed